gridded_etl_tools/utils/publish.py

```diff
@@ -60,18 +63,25 @@ def parse(self, publish_dataset: xr.Dataset):
             threads_per_worker=self.dask_num_threads,
             n_workers=self.dask_num_workers,
         ) as cluster:
-            # IPLD objects can't pickle successfully in Dask distributed schedulers so we remove the distributed client
-            # in these cases
-            with Client(cluster) if not isinstance(self.store, IPLD) else nullcontext():
+            with Client(cluster):
                 self.info(f"Dask Dashboard for this parse can be found at {cluster.dashboard_link}")
                 try:
                     # Attempt to find an existing Zarr, using the appropriate method for the store. If there is
                     # existing data and there is no rebuild requested, start an update. If there is no existing data,
                     # start an initial parse. If rebuild is requested and there is no existing data or allow overwrite
-                    # has been set, write a new Zarr, overwriting (or in the case of IPLD, not using) any existing
-                    # data. If rebuild is requested and there is existing data, but allow overwrite is not set, do not
-                    # start parsing and issue a warning.
+                    # has been set, write a new Zarr, overwriting any existing data.
+                    # If rebuild is requested and there is existing data, but allow overwrite is not set, do not start
+                    # parsing and issue a warning.
                     if self.store.has_existing and not self.rebuild_requested:
+
+                        # If zarr.json is present, the format is considered 3. Otherwise, it is considered format 2.
+                        # os.path.join works for both S3 and local.
+                        if self.store.fs().exists(os.path.join(self.store.path, "zarr.json")):
```

Comment:
Since `output_zarr3` is an init parameter I think this check should take place within `init` itself. I understand the logic of keeping it close to the actual act of publishing data but I think we should colocate parameters and their checks.